File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum = nums[0]

        cursum = 0

        for n in nums:
            if cursum < 0:
                cursum = 0

            cursum += n

            max_sum = max(cursum, max_sum)

        return max_sum

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxsum = nums[0];
        int cursum = 0;

        for (int a : nums){
            if (cursum < 0){
                cursum = 0;
            }
            cursum += a;
            maxsum = Math.max(maxsum, cursum);
        }
        return maxsum;
        
    }
}